editor/script_create_dialog.cpp

```diff
@@ -244,40 +249,46 @@ String ScriptCreateDialog::_get_class_name() const {
 }
 
 void ScriptCreateDialog::_class_name_changed(const String &p_name) {
-	if (_validate_class(class_name->get_text())) {
-		is_class_name_valid = true;
-	} else {
-		is_class_name_valid = false;
-	}
+	is_class_name_valid = _validate_class(class_name->get_text());
 	_update_dialog();
 }
 
 void ScriptCreateDialog::_parent_name_changed(const String &p_parent) {
-	if (_validate_parent(parent_name->get_text())) {
-		is_parent_name_valid = true;
-	} else {
-		is_parent_name_valid = false;
-	}
+	is_parent_name_valid = _validate_parent(parent_name->get_text());
 	_update_dialog();
 }
 
 void ScriptCreateDialog::_template_changed(int p_template) {
-	String selected_template = p_template == 0 ? "" : template_menu->get_item_text(p_template);
-	EditorSettings::get_singleton()->set_project_metadata("script_setup", "last_selected_template", selected_template);
-	if (p_template == 0) {
-		//default
-		script_template = "";
-		return;
-	}
-	int selected_id = template_menu->get_selected_id();
-
-	for (int i = 0; i < template_list.size(); i++) {
-		const ScriptTemplateInfo &sinfo = template_list[i];
-		if (sinfo.id == selected_id) {
-			script_template = sinfo.dir.plus_file(sinfo.name + "." + sinfo.extension);
-			break;
+	const ScriptLanguage::ScriptTemplate &sinfo = _get_current_template();
+	// Update last used dictionaries
+	if (is_using_templates && !parent_name->get_text().begins_with("\"res:")) {
+		if (sinfo.origin == ScriptLanguage::TemplateLocation::TEMPLATE_PROJECT) {
+			// Save the last used template for this node into the project dictionnary.
```

Comment:
```suggestion
			// Save the last used template for this node into the project dictionary.
```
Misspelled dictionary. Same below.

Reply:
Fixed. Was a mix-up with the french spelling...